Refactor docs for the new integrations page (#2290)
grafana/oncall: docs/sources/integrations/_index.md

To configure an integration for Grafana OnCall:

1. In Grafana OnCall, navigate to the **Integrations** tab and click **+ New integration**.
1. Select an integration type from the [list of available integrations]({{< relref "#list-of-available-integrations" >}}),
if the integration you want isn’t listed, then select **Webhook**.
1. Fill in title and description for your integration, assign it to the team and click **Create Integration**.

> 1. Integration page will open, you can use HTTP Endpoint url to send events from monitoring system, click **How to connect**
> link for more specific information.
> 1. Complete any necessary configurations in your tool to send alerts to Grafana OnCall.
> 1. Click **Send demo alert** to send a test alert to Grafana OnCall.

Reviewer: ```suggestion
1. The Integration page will open. Here you will see details about the Integration.
You can use the HTTP Endpoint url to send events from an external monitoring system.
Click the **How to connect** link for more information.
1. Complete any necessary configurations in your tool to send alerts to Grafana OnCall.
1. Click **Send demo alert** to send a test alert to Grafana OnCall.
```

### Complete the integration configuration

- Review and customise grouping, autoresolution, autoacknowledge, etc [templates]({{< relref "../jinja2-templating" >}})
if you want to customise alert behaviour for your team
- Review and customise [other templates]({{< relref "../jinja2-templating" >}}) to change how alert groups are displayed
in different parts of Grafana OnCall: UI, messengers, emails, notifications, etc.
- Add routes to your integration to route alerts to different users and teams based on labels or other data

> - Connect your escalation chains to the routes to notify right people at the right time

Reviewer: ```suggestion
- Connect your escalation chains to routes to notify the right people, at the right time
```

> - Learn [how to start Maintenance mode]({{< relref "#maintenance-mode" >}}) for integration

Reviewer: ```suggestion
- Learn [how to start Maintenance Mode]({{< relref "#maintenance-mode" >}}) for an integration
```

> - Send demo alerts to integration to make sure routes, templates and escalation are working as expected. Consider using

Reviewer: ```suggestion
- Send demo alerts to an integration to make sure routes, templates, and escalations, are working as expected. Consider using
```

`Debug Maintenance mode` to avoid sending real notifications to your team

### Manage integrations

To manage existing integrations, navigate to the **Integrations** tab in Grafana OnCall and select the integration
you want to manage.

> #### Maintenance mode

Reviewer: ```suggestion
#### Maintenance Mode
```

> Start maintenance mode when performing scheduled maintenance or updates on the infrastructure, which may trigger false alarms.

Reviewer: ```suggestion
Start maintenance mode when performing scheduled maintenance or updates on your infrastructure, which may trigger false alarms.
```

There are two possible maintenance modes:

- **Debug** - test routing and escalations without real notifications. Alerts will be processed as usual, but no notifications
will be sent to users.
- **Maintenance** - group alerts into one during infrastructure work.

##### Manage maintenance mode

1. Go to Integration page and click **Three dots**
1. Select **Start Maintenance Mode**
1. Select **Debug** or **Maintenance** mode
1. Set **Duration** of maintenance mode
1. Click **Start**
1. If you want to stop maintenance mode before it's end, click **Three dots** and select **Stop Maintenance Mode**

#### Heartbeat monitoring

An OnCall heartbeat acts as a healthcheck for alert group monitoring. You can configure you monitoring to regularly send alerts
to the heartbeat endpoint. If OnCall doen't receive one of these alerts, it will create an new alert group and escalate it

1. Go to Integration page and click **Three dots**
1. Select **Heartbeat Settings**
1. Set **Heartbeat interval**
1. Copy **Endpoint** into you monitoring system.

> More specific instructions can be found in specific integration documentation.

Reviewer: ```suggestion
More specific instructions can be found in a specific integration's documentation.
```

Should point to the most specific place in the alert source related to the alert group. Also rendering result will be
available in other templates as a variable `{{ source_link }}`.

#### Edit integration name, description and assigned team

To edit the name of an integration:

1. Navigate to the **Integrations** tab, select an integration from the list of enabled integrations.
1. Click the **three dots** next to the integration name and select **Integration settings**.

> 1. Provide a new name, description and team and click **Save**.

Reviewer: ```suggestion
1. Provide a new name, description, and team, and click **Save**.
```
